internal/extension/extension.go

```diff
@@ -109,7 +109,10 @@ func (em *ExtensionManager) SendStartInvocationRequest(ctx context.Context, even
 		if traceId != "" {
 			ctx = context.WithValue(ctx, DdTraceId, traceId)
 		}
-		parentId := response.Header.Get(string(DdParentId))
+		parentId := traceId
+		if pid := response.Header.Get(string(DdParentId)); pid != "" {
+			parentId = pid
+		}
```

Comment:
If the parent id is not set, later on we will throw out the found trace context completely.